update php docs (#1590)
SigNoz/signoz.io: docs/instrumentation/php.md

- [Send traces via OTel Collector binary](#send-traces-via-otel-collector-binary) (recommended)

### Send traces directly to SigNoz cloud

Here we will be sending traces to SigNoz cloud in 4 easy steps, if you want to send traces to self hosted SigNoz , you can refer to [this](https://signoz.io/docs/instrumentation/PHP/#send-traces-to-self-hosted-signoz). We will be using Zero-code configuration for Automatic Instrumentation.

**Step 1: Setup Development Environment**

To configure our PHP application to send data, we need to use OpenTelemetry PHP extension, since the extension is built from source. Install the build tools as follows :

<Tabs>
<TabItem value="linux" label="Linux (apt)" default>

```
sudo apt-get install gcc make autoconf
```

</TabItem>
<TabItem value="mac" label="Mac (apt)">

```
brew install gcc make autoconf
```

</TabItem>
</Tabs>

**Step 2: Build the extension**

With our environment set up we can install the extension using PECL.

```bash
pecl install opentelemetry
```

after successfully installing this, add the extension to `php.ini` file of your project.

```php
[opentelemetry]
extension=opentelemetry.so
```

> verify that the extension is enabled or not by running

Reviewer: ```suggestion
Verify that the extension is enabled by running:
```

```
php -m | grep opentelemetry
```

> The expected output is

Reviewer: ```suggestion
This should output:
```

```

> opentelemetry

Reviewer: Can you add an image of the output on terminal

Author: not needed, only a string will be displayed in terminal

```

**Step 3: Add the dependencies**

> Add additional dependencies required for OpenTelemetry SDK for PHP to perform automatic instrumentation using this command :

Reviewer: ```suggestion
Add additional dependencies required for OpenTelemetry SDK for PHP to perform automatic instrumentation using this command:
```

> Add additional dependencies required for OpenTelemetry SDK for PHP to perform automatic instrumentation using this command :

Reviewer: ```suggestion
Add dependencies required for OpenTelemetry SDK for PHP to perform automatic instrumentation using this command :
```

> :::info

Reviewer: Can we put this info admonition below after the code block we have below.

Author: sure

> You can install the additional dependencies by OpenTelemetry from [here](https://packagist.org/explore/?query=open-telemetry) for the PHP framework you are using!

Reviewer: ```suggestion
You can install the additional dependencies provided by OpenTelemetry for different PHP frameworks from [here](https://packagist.org/explore/?query=open-telemetry).
```

| OTEL_TRACES_EXPORTER           | otlp                                         |
| OTEL_METRICS_EXPORTER          | none                                         |
| OTEL_LOGS_EXPORTER             | none                                         |
| OTEL_EXPORTER_OTLP_PROTOCOL    | http/protobuf                                |
| OTEL_EXPORTER_OTLP_ENDPOINT    | `<SIGNOZ_ENDPOINT>` replace this with SigNoz cloud endpoint                       |
| OTEL_EXPORTER_OTLP_HEADERS     | signoz-access-token=`<INGESTION_KEY>` replace this with the ingestion key which you must have received in mail        |
| OTEL_PROPAGATORS              | baggage,tracecontext                         |
| php -S localhost:8080 app.php             | you can replace this with the run command of your PHP application                        |

### Send traces via OTel Collector binary

OTel Collector binary helps to collect logs, hostmetrics, resource and infra attributes. It is recommended to install Otel Collector binary to collect and send traces to SigNoz cloud. You can correlate signals and have rich contextual data through this way.

> :::note

Reviewer: Can we make this as Step 1 since this is a crucial step.

Author: done
